core/pool.go

```diff
@@ -9,7 +9,7 @@ import (
 type Pool struct {
 }
 
-func (p Pool) Start(engines []Engine, provider RequestProvider, metricsManager MetricsManager, config ChukonuConfig) {
+func (p Pool) Start(engines []Engine, provider RequestProvider, metricsManager MetricsManager, config ChukonuConfig, fuse chan bool, ack chan bool) {
```

Comment:
It should not be a common case, but it could happen:
The current way we are checking resource usage and spawning workers is not absolutely accurate. Even though I'm checking resource every time a goroutine is spawned, the mem usage I check at that point does not reflect the actual amount that WILL cost, because the goroutines spawned before this checkpoint may not have immediately started to work and consume resources, it totally depends on the go runtime scheduling. 
So let's say I'm setting the mem threshold to be 80%, and when spawning, once I see the mem usage is beyond this point, I stop spawning new workers, but seconds later, the mem usage could go up to 90, 95%, because goroutines are spawned so fast and the spawned goroutines may not start work immediately after they are spawned. Some of the earlier spawned goroutines only start working after the 80% point is reached...